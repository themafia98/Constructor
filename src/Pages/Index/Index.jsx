import React from 'react';
import PropTypes from 'prop-types';
import eventEmitter from '../../EventEmitter';
import {Redirect} from 'react-router-dom';

import {middlewareLogin} from '../../redux/middleware/loadUserMiddleware';

import Loader from '../../components/loading/Loader';
import Registration from '../../components/Registration/Registration';

import {connect} from 'react-redux';

import './index.scss';

class Index extends React.PureComponent {

    static propTypes = {
        config: PropTypes.object,
    }

    state = {
        title:  "Constructor",
        registrationActive: false,
        regStatus: false,
        wrongEnter: false,
        error: ''
    }

    statusRegistration = (event) => {
        event.additionalUserInfo.isNewUser ?
        this.setState({...this.state,
                        regStatus: true,
                        registrationActive: false,
                        error: 'Account create!'
                    })
        : console.log('error registration');
    }

    showBox = (event) => {
        this.setState ({
            ...this.state,
        registrationActive: this.state.registrationActive ? false : true
        })
    }

    authTo = (event) => {
            this.props.dispatch(middlewareLogin(this.emailImput.value, this.passwordImput.value));
}
    emailImput = null;
    passwordImput = null;
    emailRef = (node) => this.emailImput = node;
    passwordRef = (node) => this.passwordImput = node;

    render(){

        if (this.props.active) return <Redirect to = '/Cabinet' />
        else if (!this.props.active) {
            let currentSelected = this.state.registrationActive;
            return (
                <div className = 'LoginPage flex-column'>
                        <h1>{this.state.title}</h1>
                        <div className = 'LoginBox'>
                            <div className = 'LoginForm'>
                                <h3>Connect form</h3>
                                {
                                    this.props.wrongEnter || this.state.regStatus ?
                                    <p className = 'error'>{this.props.wrongEnter}</p>
                                    : null
                                }
                                <p>E-mail</p>
                                <input ref = {this.emailRef} type = 'text' />
                                <p>Password</p>
                                <input ref = {this.passwordRef} type = 'password' />
                                <input 
                                    onClick = {this.authTo} 
                                    className = 'loginButton'
                                    type = 'button'
                                    value = 'enter' />
                                <input
                                    onClick = {this.showBox}
                                    className = {currentSelected ? `loginButton selected`
                                        : 'loginButton'}
                                    type = 'button'
                                    value = 'registration'
                                    />
                            </div>
                        </div>
                        {
                            this.state.registrationActive ?
                            <Registration auth = {this.props.auth} />
                            : null
                        }
                </div>
            )
        } else  return <Loader path = '/img/loading.gif' type = 'session' />
    }

    componentDidMount = (e) => {
        eventEmitter.on('EventRegistrationCorrect', this.statusRegistration);
    }
    componentWillUnmount = (e) => {
        eventEmitter.off('EventRegistrationCorrect', this.statusRegistration);
    }
}

const mapStateToProps = (state) => {
    console.log('state index');
    return {
        active: state.cabinet.active, 
        wrongEnter: state.cabinet.error
    }
  }

export default connect(mapStateToProps)(Index);