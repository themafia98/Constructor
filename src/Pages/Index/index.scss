
html, body {
    height: 100%;
    min-height: 100%;
    overflow-x: hidden;
    // background-color:  rgb(195, 225, 231);
}

input {
    cursor: pointer;
    outline: none;
}

.container {
    position: relative;
    width: 100%;
    height: 100%;
    max-width: 90vw;
    margin: 0 auto;
   
}

.flex-row { 
    display: flex;
    flex-direction: row;
    align-items: center;
}


.flex-column {
    display: flex;
    flex-direction: column;
    align-items: center;
}


.col-12 {
    flex: 0 0 100%;
}

.col-6 {
    flex: 0 0 50%;
}

.LoginPage {

    z-index: -1;
    h1 {
        color: green;
    }
    .LoginBox {
        height: 400px;
        display: flex;
        background-color: lightcyan;
        justify-content: center;
        align-items: center;
        flex-direction: column;
        box-shadow: 0 0 5px;
        width: 40%;
        z-index: 100;

        .LoginForm {

            display: flex;
            flex-direction: column;
            width: 50%;

            h3 {
                margin: 0;
                text-align: center;
                color: rgb(17, 145, 17);
            }
            p {
                text-align: center;
                font-weight: bold;
                margin: 5px;
            }
            .loginButton {
                cursor: pointer;
                width: 100%;
                border: 1px solid rgb(145, 116, 116);
                border-radius: 20px;
                height: 30px;
                font-weight: bold;
                background: rgb(5, 122, 5);
                margin-top: 10px;
                margin-bottom: 10px;
                display: block;
                color: white;
                transition: .3s all linear;

            }

            .loginButton:hover {
                background: lightgreen;
                color: red;
            }

            .selected {
                color: red;
                background-color: lightgreen;
            }

            .error {
                color: red;
            }
        }
    }
}

input[type='text'], input[type='password']{
    background-color: white;
    box-shadow:  0 0 3px;
    border-radius: 10px;
}



.loaderBox {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    position: absolute;
    transform: translate(-50%,-50%);
    left: 50%;
    top: 50%;

    p {
        font-size: 20px;
        font-weight: bold;
        text-align: center;
    }

    .loader {
        width: 60%;
    }
}