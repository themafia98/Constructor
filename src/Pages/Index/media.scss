@media screen and (max-width: 760px){
    .LoginPage {
        .LoginBox {
            width: 90%;
            height: 100%;

        }
    }
    .loaderBox .loader {
        width: 75%;
    }
}