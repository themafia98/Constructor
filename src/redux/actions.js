const SAVE_CHANGES = 'SAVE_CHANGES';
const LOAD_CURRENT = 'LOAD_CURRENT';

const LOAD_USER_CABINET = 'LOAD_USER_CABINET';


const saveChangesAction = (state) => {
    return {
        type: SAVE_CHANGES,
        project: state,
    }
}

const loadCurrentProjectAction = (state) => {
    return {
        type: LOAD_CURRENT,
        currentEditable: state,
    }

}

const loadUserAction = (state) => {
    return {
        type: LOAD_USER_CABINET,
        idUser: state ? state.uid : 'NO_USER'
    }
}


export {
    SAVE_CHANGES, saveChangesAction,
    LOAD_CURRENT, loadCurrentProjectAction,
    LOAD_USER_CABINET, loadUserAction,
}