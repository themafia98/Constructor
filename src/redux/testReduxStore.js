import configureStore from 'redux-mock-store';


const initialStateOffline = {
    cabinet: {
    active: false,
    idUser: null,
    projects: [
        {
            components: [],
            id:null,
            title: null,
            type: null,
        }
    ],
    },
    builder: {}
};

const initialState = {
    cabinet: {
    active: true,
    idUser: "asdasdsdasda",
    projects: [
        {
            components: [
               {
                backgroundImage: null,
                color: null,
                content: null,
                fontSize: "135",
                image: null,
                name: "Header",
                type: "text"
                }
            ],
            id: 0,
            title: "Test1",
            type: "landing",
        }
    ],
    },
    builder: {
        loadProject: true,
        haveUpdateLoading: true,
        idProject: 0,
        typeProject: "portfolio",
        components: []
    }
};
// here it is possible to pass in any middleware if needed into //configureStore
const mockStore = configureStore();

const store = mockStore(initialState);

export default store;
export {initialStateOffline};