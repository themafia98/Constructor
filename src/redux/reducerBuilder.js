import {LOAD_CURRENT,LOAD_UPDATE_PROJECT, EXIT_PROJECT} from './actions';

const initialState = {
    loadProject: false,
    idProject: null,
    typeProject: null,
    components: []
};
export default (state = initialState, action) => {
    switch (action.type){

        case LOAD_CURRENT: {
            return {
                ...state,
                loadProject: true,
                idProject: action.id,
                typeProject: action.typeProject,
                components: [...action.components]
            }
        }

        case LOAD_UPDATE_PROJECT: {
            return {
                ...state,
                haveUpdateLoading: action.haveUpdateLoading,
            }
        }

        case EXIT_PROJECT: {
            if (action.exit)
            return {
                ...state,
                loadProject: false,
                idProject: null,
                typeProject: null,
                components: []
            }
            else return state;
        }
        default: return state;
    }
};