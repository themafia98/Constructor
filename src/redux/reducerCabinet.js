import {LOAD_USER_CABINET, LOG_OUT_CABINET, ERROR_CABINET, CREATE_PROJECT, LOAD_UPDATE_PROJECT} from './actions';

const initialState = {
    active: false,
    idUser: null,
    projects: [],
    error: null,
    //dateConnect: new Date(Date.now()).toLocaleString().replace(/\s/ig,'').split(','),
};

export default (state = initialState, action) => {
    switch(action.type) {
        case LOAD_USER_CABINET: {

            return {
                ...state,
                active: action.active,
                idUser: action.idUser,
                projects: [...action.projects]
            }
        }

        case CREATE_PROJECT: {
            return {
                ...state,
                projects: [...action.projects]
            }
        }

        case LOAD_UPDATE_PROJECT: {
            return {
                ...state,
                projects: state.projects.map(item => {
                    if (item.id === action.idProject)
                        item.component = [...action.component]
                return item;
                })
            }
        }

        case LOG_OUT_CABINET: {
            return {
                ...state,
                active: false,
                idUser: null,
                projects: []
            }
        }

        case ERROR_CABINET: {
            return {
                ...state,
                error: action.error
            }
        }
        default: return state;
    }
}

