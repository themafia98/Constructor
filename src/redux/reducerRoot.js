import {ROOT_INIT} from './actions';
const initialState = {
    test: 0,
};

export default (state = initialState, action) => {

    switch (action.type){

        case ROOT_INIT: {
            return {
                ...state,
                test: state.test + 1,
            }
        }
        default: return state;
    }
};