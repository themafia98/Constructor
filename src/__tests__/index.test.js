import React from 'react';
import firebase from 'firebase';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {initialStateOffline} from '../redux/testReduxStore';
import Index from '../Pages/Index/Index';
import { CustomConsole } from '@jest/console';

const config = require('../config.json');
const props = {
    history: {},
    match: {},
    location: {}
}

const mockStore = configureStore();
const store = mockStore(initialStateOffline);

const IndexComponent = renderer.create(
    <MemoryRouter initialEntries = {['/']}>
        <Provider store = {store}>
            <Index {...props} config = {config} firebase = {firebase} />
        </Provider>
    </MemoryRouter>
)

test("Index test", function(){

    let snapshotIndex = IndexComponent.toJSON();
    expect(snapshotIndex).toMatchSnapshot();

    const loginButton = IndexComponent.root.find(input => input.props.className === 'loginButton enterInput');
    loginButton.props.onClick({stopPropagation: () => null});

    snapshotIndex = IndexComponent.toJSON();
    expect(snapshotIndex).toMatchSnapshot();

    const regButton = IndexComponent.root.find(input => input.props.className === 'loginButton registration');
    regButton.props.onClick({stopPropagation: () => null});

    snapshotIndex = IndexComponent.toJSON();
    expect(snapshotIndex).toMatchSnapshot();

});