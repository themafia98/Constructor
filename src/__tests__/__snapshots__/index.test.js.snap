// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index test 1`] = `
<div
  className="LoginPage flex-column"
>
  <h1>
    Constructor
  </h1>
  <div
    className="LoginPage__LoginBox"
  >
    <h3
      className="sc-AykKC cSQYJJ"
      data-titlebuild={true}
      dir="up"
    >
      B
    </h3>
    <form
      className="LoginBox__LoginForm"
    >
      <h3>
        Connect form
      </h3>
      <p>
        E-mail
      </p>
      <input
        autoComplete="off"
        placeholder="email"
        type="text"
      />
      <p>
        Password
      </p>
      <input
        autoComplete="off"
        placeholder="password"
        type="password"
      />
      <input
        className="LoginForm__loginButton enterInput"
        onClick={[Function]}
        type="button"
        value="enter"
      />
      <input
        className="LoginForm__loginButton registration"
        onClick={[Function]}
        type="button"
        value="registration"
      />
    </form>
  </div>
</div>
`;

exports[`Index test 2`] = `
<div
  className="LoginPage flex-column"
>
  <h1>
    Constructor
  </h1>
  <div
    className="LoginPage__LoginBox"
  >
    <h3
      className="sc-AykKC cSQYJJ"
      data-titlebuild={true}
      dir="up"
    >
      B
    </h3>
    <form
      className="LoginBox__LoginForm"
    >
      <h3>
        Connect form
      </h3>
      <p>
        E-mail
      </p>
      <input
        autoComplete="off"
        placeholder="email"
        type="text"
      />
      <p>
        Password
      </p>
      <input
        autoComplete="off"
        placeholder="password"
        type="password"
      />
      <input
        className="LoginForm__loginButton enterInput"
        onClick={[Function]}
        type="button"
        value="enter"
      />
      <input
        className="LoginForm__loginButton registration"
        onClick={[Function]}
        type="button"
        value="registration"
      />
    </form>
  </div>
</div>
`;
